### ISSUE TITLE
Add graph to webapp for carbon emissions by power plant types

### ISSUE THREAD

**Issue by oashmoore** [Energy-Innovation/eps-us#139]
It would be helpful to have a wedge graph (similar to the elec generation and capacity graphs) that shows CO2e emissions by power plant type over time, including imported electricity emissions if they are included in the model.

**Comment by jrissman** [comment]
Good idea!  This one won't take long to add.  I can do this tomorrow and we can include it in 3.1.2 if you like (and you don't mind making some small, easy updates to India and any other regions that will be launching on 3.1.2).

**Comment by oashmoore** [comment]
Thanks Jeff! would it be possible to do it for 3.1.1 as well? The state models are on 3.1.1

**Comment by jrissman** [comment]
3.1.1 was already released.  We can't change a release in the past.  We can only include features and fixes in new releases.

You can, of course, update the state models from 3.1.1 to 3.1.2 so they get the new features.  In the case of moving from 3.1.1 to 3.1.2, that requires a few changes to each model's input data, mostly related to breaking out coal mining from oil and gas extraction in the ISIC codes.  That's already done for the U.S. national model, so if you typically populate these variables by downscaling the national data, it might be pretty straightforward.  This would improve those models' accuracy with regard to jobs, GDP, and employee compensation impacts in these industries.

**Comment by oashmoore** [comment]
Okay, good to know! @robbieorvis what do you think?

**Comment by jrissman** [comment]
(If we don't include this new graph in 3.1.2, I'll include it in 3.2.0, which is the release that will have 25 different Industry Categories.)

**Comment by robbieorvis** [comment]
Thanks both.

Olivia, I think it’s okay if we don’t include this for Minnesota. For our purposes of creating the research note, we can calculate this in Excel rather easily.

If it were totally urgent (and I don’t think it is) we could of course stash the changes to 3.1.2 for now, implement this update, the apply the stashed changes to a 3.1.3 release. That would let us simply add this graph without dealing with the other issues.

We are of course eager to get models onto the beta 3.1.2 with the ISIC 05 06 breakout, it’s just that it would require rerunning the data export, which we don’t want to do at this time for the state models already published, though we will do this at a later date.

Olivia, we should get in touch with Jun about implementing the 05 06 split though for the new datasets so we can deploy them on 3.1.2. We are just waiting on Megan to finish updating the 1.5 scenario before officially launching 3.1.2, which I expect we can do early next week before we are ready to launch Colorado or Nevada.

________________________________
Robbie Orvis
Director of Energy Policy Design
Phone: 415-799-2171
98 Battery Street, Suite 202
San Francisco, CA  94111
www.energyinnovation.org<http://www.energyinnovation.org/>
[cid:image001.jpg@01D6EFD7.C896A010]
________________________________

Check out our new book, Designing Climate Solutions: A Policy Guide for Low-Carbon Energy<https://www.amazon.com/Designing-Climate-Solutions-Policy-Low-Carbon/dp/1610919564>
Available wherever books are sold

[Policy Design book cover]

**Comment by jrissman** [comment]
On the topic of having Jun learn new ISIC code splits, there are more splits coming in 3.2.0, and I think we might be on a **final** set of ISIC codes at that point.  I'm working on these splits before adding Industry Categories, and the splits may be done as soon as today.  So perhaps wait for that before having Jun adapt the procedures for filling out ISIC code variables.  This change will be separable from the Industry Category expansion and could even be released as a 3.1.3 if you need it published separately from the Industry Category expansion for some reason.

It also might be worth considering just waiting and moving all the state versions to 3.2 rather than 3.1.2 because of the large improvements to the Industry Sector in 3.2.

I'm going to defer work on the issue of the new graph in this issue, #139, to 3.2, since it sounds like you want to finalize and release 3.1.2 soon.

**Comment by jrissman** [comment]
Completed in commits e0b7514 and e45c9b3.

**Comment by jrissman** [comment]
Note that the new graph correctly respects the control settings that can optionally include emissions associated with imported electricity, and can exclude emissions associated with exported electricity.  The new graph shows only the emissions flagged for inclusion based on these two control levers.
